OpenRA.Game/Network/Connection.cs

```diff
@@ -87,20 +87,12 @@ protected virtual void Send(byte[] packet)
 
 		protected void AddPacket(ReceivedPacket packet)
 		{
-			lock (receivedPackets)
-				receivedPackets.Add(packet);
+			receivedPackets.Enqueue(packet);
 		}
 
 		public virtual void Receive(Action<int, byte[]> packetFn)
 		{
-			ReceivedPacket[] packets;
-			lock (receivedPackets)
-			{
-				packets = receivedPackets.ToArray();
-				receivedPackets.Clear();
-			}
-
-			foreach (var p in packets)
+			while (receivedPackets.TryDequeue(out var p))
```

Comment:
I don't understand this change, enqueueing and dequeuing might be thread safe itself, but this doesn't appear to stop an enqueue from happening between dequeues here?

Reply:
Ok never mind, this is a FIFO queue so the order won't change.